### PR TITLE
Feature/expand clientcore api

### PR DESCRIPTION
**Pull request by taki9**
- New methods on repository to handle preview documents and their comments
- New methods on repository to handle allowed child types
- New methods on repository to get metadata and value of a property of content

### CODE REVIEW

File: packages/sn-client-core/src/Repository/Preview.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      responseBody.PreviewImageUrl = `${options.document.hostName}${responseBody.PreviewAvailable}`
+      responseBody.ThumbnailImageUrl = `${options.document.hostName}${responseBody.PreviewAvailable.replace(
+        'preview',
+        'thumbnail',
+      )}`
+      return responseBody as PreviewImageData
+    }
+  }
+
+  public async getExisting(options: { document: DocumentData; version: string; abortController?: AbortController }) {
```
Comment (herflis) [inline packages/sn-client-core/src/Repository/Preview.ts:45]: I would rather name this method to `getExistingImages`

File: packages/sn-client-core/test/allowed-child-types.test.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
+import { ContentType } from '@sensenet/default-content-types'
+import { Repository } from '../src/Repository'
+import { ODataCollectionResponse } from '../src/Models'
+
+describe('AllowedChildTyoes', () => {
```
Comment (herflis) [inline packages/sn-client-core/test/allowed-child-types.test.ts:5]:
```suggestion
describe('AllowedChildTypes', () => {
```

File: packages/sn-client-core/test/allowed-child-types.test.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+          d: { results: [], __count: 0 },
+        } as ODataCollectionResponse<ContentType>
+      }
+      const response = await repository.allowedChildTypes.getImplicit({
+        idOrPath: 'Root/Sites/Default_Site',
+      })
+      expect(response.d).toEqual({ results: [], __count: 0 })
+    })
+
+    it('should throw on unsuccessfull request', (done) => {
```
Comment (herflis) [inline packages/sn-client-core/test/allowed-child-types.test.ts:38]:
```suggestion
    it('should throw on unsuccessful request', (done) => {
```

File: packages/sn-client-core/test/allowed-child-types.test.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+          d: { results: [], __count: 0 },
+        } as ODataCollectionResponse<ContentType>
+      }
+      const response = await repository.allowedChildTypes.getExplicit({
+        idOrPath: 'Root/Sites/Default_Site',
+      })
+      expect(response.d).toEqual({ results: [], __count: 0 })
+    })
+
+    it('should throw on unsuccessfull request', (done) => {
```
Comment (herflis) [inline packages/sn-client-core/test/allowed-child-types.test.ts:69]:
```suggestion
    it('should throw on unsuccessful request', (done) => {
```

File: packages/sn-client-core/test/allowed-child-types.test.ts [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+      ;(mockResponse as any).ok = true
+      mockResponse.json = async () => {
+        return []
+      }
+      const response = await repository.allowedChildTypes.get({
+        idOrPath: 'Root/Sites/Default_Site',
+      })
+      expect(response).toEqual({ d: { __count: 0, results: [] } })
+    })
+    it('should throw on unsuccessfull request', (done) => {
```
Comment (herflis) [inline packages/sn-client-core/test/allowed-child-types.test.ts:97]:
```suggestion
    it('should throw on unsuccessful request', (done) => {
```

File: packages/sn-client-core/test/allowed-child-types.test.ts [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+  const fetchMock: Repository['fetchMethod'] = async () => {
+    return mockResponse
+  }
+
+  beforeEach(() => {
+    repository = new Repository(undefined, fetchMock)
+  })
+
+  afterEach(() => {
+    repository.dispose()
```
Comment (zoltanbedi) [inline packages/sn-client-core/test/allowed-child-types.test.ts:21]: Just a heads up. This dispose method does nothing.

File: packages/sn-document-viewer-react/src/components/page-widgets/style.ts [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import styled from 'styled-components'
-import { DraftCommentMarker } from '../../models'
+import { DraftCommentMarker } from '@sensenet/client-core'
```
Comment (zoltanbedi) [inline packages/sn-document-viewer-react/src/components/page-widgets/style.ts:2]: This type should be left here. It is only used in one of the components state.

**Pull request merged**